scripts/detect-ec2-spot-preemption.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+while true
+do
+    # This IP address comes from:
+    # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-interruptions.html#spot-instance-termination-notices
+    # It's a special endpoint set up by AWS whereby AWS instances can view metadata about themselves.
+    # One such piece of metadata is the termination time, which is only set when the spot instance is to be
+    # pre-empted (you get a 404 otherwise).
+    # This script was partially taken from https://stackoverflow.com/q/32613600/14089059 .
+    if [ -z $(curl -Is http://169.254.169.254/latest/meta-data/spot/termination-time | head -1 | grep 404 | cut -d \  -f 2) ]
```

Comment:
Could we use the instance-action endpoint instead? It might make the code a bit cleaner and seems to be the recommended way to do it:

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-interruptions.html#spot-instance-termination-notices